src/System.IO.Pipes/src/System/IO/Pipes/NamedPipeClientStream.Windows.cs

```diff
@@ -129,6 +134,19 @@ public int NumberOfServerInstances
             }
         }
 
+        private void ValidateRemotePipeIsCurrentUser()
+        {
+            SecurityIdentifier currentUserSid = GetCurrentUser();
+            PipeSecurity accessControl = this.GetAccessControl();
+
+            IdentityReference remoteOwnerSid = accessControl.GetOwner(typeof(SecurityIdentifier));
+            if (remoteOwnerSid != currentUserSid)
+            {
+                State = PipeState.Closed;
+                throw new UnauthorizedAccessException();
```

Comment:
You might want a custom message eg "Could not connect to the pipe because it was not owned by the current user."